crates/proof-of-sql/src/base/database/order_by_util.rs

```diff
@@ -31,6 +31,65 @@ pub(crate) fn compare_indexes_by_columns<S: Scalar>(
         .unwrap_or(Ordering::Equal)
 }
 
+/// Compares the tuples `(left[0][i], left[1][i], ...)` and
+/// `(right[0][j], right[1][j], ...)` in lexicographic order.
+///
+/// Requires that columns in `left` and `right` have the same column types for now
+///
+/// # Panics
+/// Panics if `left` and `right` have different number of columns
+/// which should never happen since this function should only be called
+/// for joins.
+#[allow(dead_code)]
+pub(crate) fn compare_indexes_of_tables_by_columns<S: Scalar>(
+    left: &[Column<S>],
+    right: &[Column<S>],
+    i: usize,
+    j: usize,
```

Comment:
NIT: I think the function name or variable names could more strongly imply that the comparison is for a single row in each table. Maybe something like `fn compare_single_row_of_tables`, `left_row_index`, and `right_row_index`?